File: DeliveryCrab/Models/Product.cs
namespace DeliveryCrab.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Image { get; set; } //ссылка на картинку
        public float Price { get; set; }
        public string Description { get; set; }

        public int RestaurantId { get; set; } // внешний ключ
        public Restaurant? Restaurant { get; set; }  // навигационное свойство(добавляет каскадное удаление)
        
    }
}

File: DeliveryCrab/ClientApp/src/app/models/product.ts
import { Restaurant } from "./restaurant";

export class Product{
  constructor(
    public id?:number,
    public name?:string,
    public image?:string,
    public price?:number,
    public description?:string,
    public restaurantid?:number,
    public restaurant?:Restaurant
  ){}

}
